publisher/publisher.go

```diff
@@ -254,7 +255,8 @@ func (pub *Impl) SubmitToSingleCTWithResult(ctx context.Context, req *pubpb.Requ
 			return nil, err
 		}
 		var body string
-		if respErr, ok := err.(jsonclient.RspError); ok && respErr.StatusCode < 500 {
+		var respErr jsonclient.RspError
```

Comment:
Nit: below you call a very similar variable `rspError`. Since these are both taking a value of type `<somethin>.RspError`, I think consolidating on the name `rspError` is good.